R/functions_netCDF.R

```diff
@@ -533,11 +1096,127 @@ read_netCDF_to_raster <- function(x, ...) {
   r
 }
 
+#' Read data from a netcdf into an array or matrix
+#'
+#' @param x a netCDF fiile
+#' @return an array
+#' @export
+#' 
+read_netCDF_to_array <- function(x, locations) {
+  
+  # open file, writeable --------------
+  nc <- ncdf4::nc_open(x, write = TRUE)
+  on.exit(ncdf4::nc_close(nc))
+  
+  # figure out dimensionality
+  nc_dims <-  attributes(nc$dim)$names #dims of netcdf
+  
+  if('site' %in% nc_dims) isGridded = FALSE
+  if('lat' %in% nc_dims) isGridded = TRUE
+  
+  has_T_timeAxis <-  ifelse('time' %in% nc_dims, TRUE, FALSE)
```

Comment:
lines 1117 and 1118 (equivalently): --> `has_T_timeAxis <- "time" %in% nc_dims`

Reply:
also, see comment to line 480: the names of the time and vertical dimensions are user specified; however, here we assume a specific name. This doesn't work when a user names the vertical dimension for instance "height".
--> I think the best option is to fix the names of the these two dimensions as we do for longitude and latitude